In [ ]:
import pandas as pd
import numpy as np

In [4]:
data=pd.read_csv('C:/Users/aditya/Desktop/Internship/Datasets/data1_nonull.csv')

In [ ]:
data.head()

In [5]:
data.shape

(77389, 65)

# 1. The World Bank's international debt data (data loading done)



# Finding the number of distinct countries

In [6]:
data['Country Name'].unique

<bound method Series.unique of 0                                              Afghanistan
1                                              Afghanistan
2                                              Afghanistan
3                                              Afghanistan
4                                              Afghanistan
                               ...                        
77384                                                  NaN
77385                                                  NaN
77386                                                  NaN
77387    Data from database: International Debt Statistics
77388                             Last Updated: 10/10/2021
Name: Country Name, Length: 77389, dtype: object>

In [7]:
data.tail()

Country Name Country Code  \
77384                                                NaN          NaN   
77385                                                NaN          NaN   
77386                                                NaN          NaN   
77387  Data from database: International Debt Statistics          NaN   
77388                           Last Updated: 10/10/2021          NaN   

      Counterpart-Area Name Counterpart-Area Code Series Name Series Code  \
77384                   NaN                   NaN         NaN         NaN   
77385                   NaN                   NaN         NaN         NaN   
77386                   NaN                   NaN         NaN         NaN   
77387                   NaN                   NaN         NaN         NaN   
77388                   NaN                   NaN         NaN         NaN   

       1970  1971  1972  1973  ...  2019  2020  2021  2022  2023  2024  2025  \
77384   NaN   NaN   NaN   NaN  ...   NaN   NaN   NaN   NaN   NaN   NaN   NaN   
77385   NaN   NaN   NaN   NaN  ...   NaN   NaN   NaN   NaN   NaN   NaN   NaN   
77386   NaN   NaN   NaN   NaN  ...   NaN   NaN   NaN   NaN   NaN   NaN   NaN   
77387   NaN   NaN   NaN   NaN  ...   NaN   NaN   NaN   NaN   NaN   NaN   NaN   
77388   NaN   NaN   NaN   NaN  ...   NaN   NaN   NaN   NaN   NaN   NaN   NaN   

       2026  2027  2028  
77384   NaN   NaN   NaN  
77385   NaN   NaN   NaN  
77386   NaN   NaN   NaN  
77387   NaN   NaN   NaN  
77388   NaN   NaN   NaN  

[5 rows x 65 columns]

In [8]:
data.dropna(inplace=True)

In [9]:
data.shape

(5635, 65)

In [12]:
a=data['Country Name'].unique()

In [13]:
len(a)

132

# SO THERE  UNIQUE 132 COUNTRIES IN TOTAL

#  Finding out the distinct debt indicators


In [16]:
a=data['Series Code'].unique()

In [17]:
len(a)

127

# SO THERE ARE 127 DISTINCT DEBT INDICATORS

# 4. Totaling the amount of debt owed by the countries

In [18]:
data.head()

Country Name Country Code Counterpart-Area Name Counterpart-Area Code  \
110  Afghanistan          AFG                 World                   WLD   
115  Afghanistan          AFG                 World                   WLD   
338  Afghanistan          AFG                 World                   WLD   
377  Afghanistan          AFG                 World                   WLD   
389  Afghanistan          AFG                 World                   WLD   

                                           Series Name     Series Code  1970  \
110  Disbursements on external debt, long-term (DIS...  DT.DIS.DLXF.CD   0.0   
115  Disbursements on external debt, public and pub...  DT.DIS.DPPG.CD   0.0   
338                  PPG, bilateral (DIS, current US$)  DT.DIS.BLAT.CD   0.0   
377               PPG, multilateral (DIS, current US$)  DT.DIS.MLAT.CD   0.0   
389         PPG, official creditors (DIS, current US$)  DT.DIS.OFFT.CD   0.0   

     1971  1972  1973  ...        2019      2020        2021         2022  \
110   0.0   0.0   0.0  ...  28590201.1  440209.2  91525581.2  109228850.6   
115   0.0   0.0   0.0  ...  28590201.1  440209.2  91525581.2  109228850.6   
338   0.0   0.0   0.0  ...  27851000.5       0.0  81807974.7   86924699.4   
377   0.0   0.0   0.0  ...    739200.6  440209.2   9717606.5   22304151.2   
389   0.0   0.0   0.0  ...  28590201.1  440209.2  91525581.2  109228850.6   

           2023        2024        2025        2026       2027       2028  
110  76137774.7  49745910.6  29280699.4  15282698.8  9068511.5  3491523.9  
115  76137774.7  49745910.6  29280699.4  15282698.8  9068511.5  3491523.9  
338  59607890.3  37990151.3  21443526.5   9922365.3  6005463.7  1960000.0  
377  16529884.4  11755759.3   7837172.9   5360333.5  3063047.8  1531523.9  
389  76137774.7  49745910.6  29280699.4  15282698.8  9068511.5  3491523.9  

[5 rows x 65 columns]

In [19]:
data.columns

Index(['Country Name', 'Country Code', 'Counterpart-Area Name',
       'Counterpart-Area Code', 'Series Name', 'Series Code', '1970', '1971',
       '1972', '1973', '1974', '1975', '1976', '1977', '1978', '1979', '1980',
       '1981', '1982', '1983', '1984', '1985', '1986', '1987', '1988', '1989',
       '1990', '1991', '1992', '1993', '1994', '1995', '1996', '1997', '1998',
       '1999', '2000', '2001', '2002', '2003', '2004', '2005', '2006', '2007',
       '2008', '2009', '2010', '2011', '2012', '2013', '2014', '2015', '2016',
       '2017', '2018', '2019', '2020', '2021', '2022', '2023', '2024', '2025',
       '2026', '2027', '2028'],
      dtype='object')

In [21]:
data.drop(['2022', '2023', '2024', '2025','2026', '2027', '2028'],inplace=True,axis=1)

In [23]:
data.columns

Index(['Country Name', 'Country Code', 'Counterpart-Area Name',
       'Counterpart-Area Code', 'Series Name', 'Series Code', '1970', '1971',
       '1972', '1973', '1974', '1975', '1976', '1977', '1978', '1979', '1980',
       '1981', '1982', '1983', '1984', '1985', '1986', '1987', '1988', '1989',
       '1990', '1991', '1992', '1993', '1994', '1995', '1996', '1997', '1998',
       '1999', '2000', '2001', '2002', '2003', '2004', '2005', '2006', '2007',
       '2008', '2009', '2010', '2011', '2012', '2013', '2014', '2015', '2016',
       '2017', '2018', '2019', '2020', '2021'],
      dtype='object')

In [24]:
data.dtypes

Country Name              object
Country Code              object
Counterpart-Area Name     object
Counterpart-Area Code     object
Series Name               object
Series Code               object
1970                     float64
1971                     float64
1972                     float64
1973                     float64
1974                     float64
1975                     float64
1976                     float64
1977                     float64
1978                     float64
1979                     float64
1980                     float64
1981                     float64
1982                     float64
1983                     float64
1984                     float64
1985                     float64
1986                     float64
1987                     float64
1988                     float64
1989                     float64
1990                     float64
1991                     float64
1992                     float64
1993                     float64
1994      

In [27]:
data_1=data.select_dtypes('float64')

In [29]:
data_1.head()

1970  1971  1972  1973  1974  1975  1976  1977  1978  1979  ...  \
110   0.0   0.0   0.0   0.0   0.0   0.0   0.0   0.0   0.0   0.0  ...   
115   0.0   0.0   0.0   0.0   0.0   0.0   0.0   0.0   0.0   0.0  ...   
338   0.0   0.0   0.0   0.0   0.0   0.0   0.0   0.0   0.0   0.0  ...   
377   0.0   0.0   0.0   0.0   0.0   0.0   0.0   0.0   0.0   0.0  ...   
389   0.0   0.0   0.0   0.0   0.0   0.0   0.0   0.0   0.0   0.0  ...   

           2012        2013        2014        2015        2016       2017  \
110  52059123.0  36656261.1  31740710.4  26037493.8  11669646.0  3241563.2   
115  52059123.0  36656261.1  31740710.4  26037493.8  11669646.0  3241563.2   
338         0.0         0.0  10120533.3  17992800.0   1438712.0  2082133.3   
377  52059123.0  36656261.1  21620177.1   8044693.8  10230934.0  1159429.9   
389  52059123.0  36656261.1  31740710.4  26037493.8  11669646.0  3241563.2   

           2018        2019      2020        2021  
110  16323196.4  28590201.1  440209.2  91525581.2  
115  16323196.4  28590201.1  440209.2  91525581.2  
338  10346133.3  27851000.5       0.0  81807974.7  
377   5977063.1    739200.6  440209.2   9717606.5  
389  16323196.4  28590201.1  440209.2  91525581.2  

[5 rows x 52 columns]

In [30]:
data['Sum']=data_1.sum(axis=1)

In [31]:
data.head()

Country Name Country Code Counterpart-Area Name Counterpart-Area Code  \
110  Afghanistan          AFG                 World                   WLD   
115  Afghanistan          AFG                 World                   WLD   
338  Afghanistan          AFG                 World                   WLD   
377  Afghanistan          AFG                 World                   WLD   
389  Afghanistan          AFG                 World                   WLD   

                                           Series Name     Series Code  1970  \
110  Disbursements on external debt, long-term (DIS...  DT.DIS.DLXF.CD   0.0   
115  Disbursements on external debt, public and pub...  DT.DIS.DPPG.CD   0.0   
338                  PPG, bilateral (DIS, current US$)  DT.DIS.BLAT.CD   0.0   
377               PPG, multilateral (DIS, current US$)  DT.DIS.MLAT.CD   0.0   
389         PPG, official creditors (DIS, current US$)  DT.DIS.OFFT.CD   0.0   

     1971  1972  1973  ...        2013        2014        2015        2016  \
110   0.0   0.0   0.0  ...  36656261.1  31740710.4  26037493.8  11669646.0   
115   0.0   0.0   0.0  ...  36656261.1  31740710.4  26037493.8  11669646.0   
338   0.0   0.0   0.0  ...         0.0  10120533.3  17992800.0   1438712.0   
377   0.0   0.0   0.0  ...  36656261.1  21620177.1   8044693.8  10230934.0   
389   0.0   0.0   0.0  ...  36656261.1  31740710.4  26037493.8  11669646.0   

          2017        2018        2019      2020        2021          Sum  
110  3241563.2  16323196.4  28590201.1  440209.2  91525581.2  935116905.3  
115  3241563.2  16323196.4  28590201.1  440209.2  91525581.2  935116905.3  
338  2082133.3  10346133.3  27851000.5       0.0  81807974.7  198840347.3  
377  1159429.9   5977063.1    739200.6  440209.2   9717606.5  736276558.0  
389  3241563.2  16323196.4  28590201.1  440209.2  91525581.2  935116905.3  

[5 rows x 59 columns]

In [33]:
a=pd.DataFrame(data.groupby('Country Name')['Sum'].sum())

In [34]:
a

Sum
Country Name               
Afghanistan    3.740468e+09
Albania        3.285194e+10
Algeria        1.981965e+12
Angola         3.801735e+11
Argentina      7.301023e+12
...                     ...
Venezuela, RB  4.193906e+12
Vietnam        4.002581e+11
Yemen, Rep.    4.739058e+10
Zambia         3.200673e+11
Zimbabwe       1.894398e+11

[132 rows x 1 columns]

# The above table shows the total debt owned country wise

# 5. Country with the highest debt

In [39]:
test = a.sort_values(['Sum'], ascending=[False])

In [40]:
test

Sum
Country Name                                                   
Low & middle income                                2.316020e+14
Middle income                                      2.284876e+14
Upper middle income                                1.590440e+14
Latin America & Caribbean (excluding high income)  7.355849e+13
Lower middle income                                6.944369e+13
...                                                         ...
Vanuatu                                            1.669289e+09
Solomon Islands                                    1.570382e+09
Somalia                                            1.215809e+09
Timor-Leste                                        1.192817e+09
Tonga                                              8.715298e+08

[132 rows x 1 columns]

In [48]:
test [0:20]

Sum
Country Name                                                    
Low & middle income                                 2.316020e+14
Middle income                                       2.284876e+14
Upper middle income                                 1.590440e+14
Latin America & Caribbean (excluding high income)   7.355849e+13
Lower middle income                                 6.944369e+13
Europe & Central Asia (excluding high income)       5.396891e+13
East Asia & Pacific (excluding high income)         4.993078e+13
Brazil                                              2.329095e+13
Mexico                                              2.203544e+13
South Asia                                          1.859534e+13
Sub-Saharan Africa (excluding high income)          1.655276e+13
Middle East & North Africa (excluding high income)  1.590150e+13
Indonesia                                           1.324773e+13
Turkey                                              1.311676e+13
India                                               1.296307e+13
IDA only                                            9.069448e+12
Least developed countries: UN classification        8.823511e+12
Argentina                                           7.301023e+12
Colombia                                            4.965788e+12
Philippines                                         4.456850e+12

# So low and middle income country has higest debt (with grouping)
# Brazil As a country has higest debt (without Grouping)

# OPENING THE CLEAN DATA AFTER REMOVAL TO NaN

In [59]:
data=pd.read_csv('C:/Users/aditya/Desktop/Internship/Datasets/data1_nonull.csv')

In [61]:
data.head()

Unnamed: 0 Country Name Country Code Counterpart-Area Name  \
0         110  Afghanistan          AFG                 World   
1         115  Afghanistan          AFG                 World   
2         338  Afghanistan          AFG                 World   
3         377  Afghanistan          AFG                 World   
4         389  Afghanistan          AFG                 World   

  Counterpart-Area Code                                        Series Name  \
0                   WLD  Disbursements on external debt, long-term (DIS...   
1                   WLD  Disbursements on external debt, public and pub...   
2                   WLD                  PPG, bilateral (DIS, current US$)   
3                   WLD               PPG, multilateral (DIS, current US$)   
4                   WLD         PPG, official creditors (DIS, current US$)   

      Series Code  1970  1971  1972  ...        2019      2020        2021  \
0  DT.DIS.DLXF.CD   0.0   0.0   0.0  ...  28590201.1  440209.2  91525581.2   
1  DT.DIS.DPPG.CD   0.0   0.0   0.0  ...  28590201.1  440209.2  91525581.2   
2  DT.DIS.BLAT.CD   0.0   0.0   0.0  ...  27851000.5       0.0  81807974.7   
3  DT.DIS.MLAT.CD   0.0   0.0   0.0  ...    739200.6  440209.2   9717606.5   
4  DT.DIS.OFFT.CD   0.0   0.0   0.0  ...  28590201.1  440209.2  91525581.2   

          2022        2023        2024        2025        2026       2027  \
0  109228850.6  76137774.7  49745910.6  29280699.4  15282698.8  9068511.5   
1  109228850.6  76137774.7  49745910.6  29280699.4  15282698.8  9068511.5   
2   86924699.4  59607890.3  37990151.3  21443526.5   9922365.3  6005463.7   
3   22304151.2  16529884.4  11755759.3   7837172.9   5360333.5  3063047.8   
4  109228850.6  76137774.7  49745910.6  29280699.4  15282698.8  9068511.5   

        2028  
0  3491523.9  
1  3491523.9  
2  1960000.0  
3  1531523.9  
4  3491523.9  

[5 rows x 66 columns]

In [62]:
data_1=data.select_dtypes('float64')

In [63]:
data['Sum']=data_1.sum(axis=1)

In [64]:
data.head()

Unnamed: 0 Country Name Country Code Counterpart-Area Name  \
0         110  Afghanistan          AFG                 World   
1         115  Afghanistan          AFG                 World   
2         338  Afghanistan          AFG                 World   
3         377  Afghanistan          AFG                 World   
4         389  Afghanistan          AFG                 World   

  Counterpart-Area Code                                        Series Name  \
0                   WLD  Disbursements on external debt, long-term (DIS...   
1                   WLD  Disbursements on external debt, public and pub...   
2                   WLD                  PPG, bilateral (DIS, current US$)   
3                   WLD               PPG, multilateral (DIS, current US$)   
4                   WLD         PPG, official creditors (DIS, current US$)   

      Series Code  1970  1971  1972  ...      2020        2021         2022  \
0  DT.DIS.DLXF.CD   0.0   0.0   0.0  ...  440209.2  91525581.2  109228850.6   
1  DT.DIS.DPPG.CD   0.0   0.0   0.0  ...  440209.2  91525581.2  109228850.6   
2  DT.DIS.BLAT.CD   0.0   0.0   0.0  ...       0.0  81807974.7   86924699.4   
3  DT.DIS.MLAT.CD   0.0   0.0   0.0  ...  440209.2   9717606.5   22304151.2   
4  DT.DIS.OFFT.CD   0.0   0.0   0.0  ...  440209.2  91525581.2  109228850.6   

         2023        2024        2025        2026       2027       2028  \
0  76137774.7  49745910.6  29280699.4  15282698.8  9068511.5  3491523.9   
1  76137774.7  49745910.6  29280699.4  15282698.8  9068511.5  3491523.9   
2  59607890.3  37990151.3  21443526.5   9922365.3  6005463.7  1960000.0   
3  16529884.4  11755759.3   7837172.9   5360333.5  3063047.8  1531523.9   
4  76137774.7  49745910.6  29280699.4  15282698.8  9068511.5  3491523.9   

            Sum  
0  1.227353e+09  
1  1.227353e+09  
2  4.226944e+08  
3  8.046584e+08  
4  1.227353e+09  

[5 rows x 67 columns]

In [68]:
a=pd.DataFrame(data.groupby('Country Name')['Sum'].sum())

In [69]:
a

Sum
Country Name               
Afghanistan    4.909411e+09
Albania        3.575531e+10
Algeria        1.995147e+12
Angola         4.022688e+11
Argentina      8.194342e+12
...                     ...
Venezuela, RB  5.185640e+12
Vietnam        4.613298e+11
Yemen, Rep.    5.100057e+10
Zambia         5.279659e+11
Zimbabwe       2.238202e+11

[132 rows x 1 columns]

In [75]:
b=a.sort_values(by=['Sum'],ascending=[False])

In [76]:
b

Sum
Country Name                                                   
Low & middle income                                2.913150e+14
Middle income                                      2.867447e+14
Upper middle income                                1.956076e+14
Lower middle income                                9.113714e+13
Latin America & Caribbean (excluding high income)  8.838561e+13
...                                                         ...
Somalia                                            2.025441e+09
Vanuatu                                            1.982464e+09
Timor-Leste                                        1.935940e+09
Solomon Islands                                    1.935556e+09
Tonga                                              8.912793e+08

[132 rows x 1 columns]

In [77]:
c=b[0:10]

# TOP 10 COUNTRIES WITH HIGEST DEPT (GROUPING)

In [78]:
import matplotlib.pyplot as plt
import seaborn as sns
%matplotlib inline
from plotly.offline import iplot
import plotly as py
import cufflinks as cf
py.offline.init_notebook_mode(connected=True)
cf.go_offline()

In [81]:
c

Sum
Country Name                                                   
Low & middle income                                2.913150e+14
Middle income                                      2.867447e+14
Upper middle income                                1.956076e+14
Lower middle income                                9.113714e+13
Latin America & Caribbean (excluding high income)  8.838561e+13
Europe & Central Asia (excluding high income)      6.641182e+13
East Asia & Pacific (excluding high income)        6.435124e+13
Brazil                                             2.737384e+13
Mexico                                             2.656635e+13
South Asia                                         2.541775e+13

In [82]:
c.iplot(kind='bar',y='Sum',title='TOP 10 COUNTRIES WITH HIGEST DEPT')

# NOW WE WISH TO FIND COUNTRIES WITH TOP 10  COUNTRIES WITH HIGHEST DEPT WITHOUT GROUPING

In [108]:
b[0:20]

Sum
Country Name                                                    
Low & middle income                                 2.913139e+14
Middle income                                       2.867600e+14
Upper middle income                                 1.956189e+14
Lower middle income                                 9.113450e+13
Latin America & Caribbean (excluding high income)   8.839174e+13
Europe & Central Asia (excluding high income)       6.641247e+13
East Asia & Pacific (excluding high income)         6.434991e+13
Brazil                                              2.737170e+13
Mexico                                              2.656635e+13
South Asia                                          2.541772e+13
Sub-Saharan Africa (excluding high income)          2.297785e+13
Middle East & North Africa (excluding high income)  1.905964e+13
India                                               1.698157e+13
Indonesia                                           1.631778e+13
Turkey                                              1.557336e+13
IDA only                                            1.383794e+13
Least developed countries: UN classification        1.350977e+13
Argentina                                           8.194342e+12
Colombia                                            6.167302e+12
Venezuela, RB                                       5.185640e+12

# LETS CREATE HAVE A NEW DATA FRAME WHEREIN WE ONLY WOULD COUNTRIES (WHICH ARE NOT GROUPED)

In [128]:
data=pd.read_csv('C:/Users/aditya/Desktop/Internship/Datasets/data1_nonull.csv')

In [129]:
data.head()

Unnamed: 0 Country Name Country Code Counterpart-Area Name  \
0         110  Afghanistan          AFG                 World   
1         115  Afghanistan          AFG                 World   
2         338  Afghanistan          AFG                 World   
3         377  Afghanistan          AFG                 World   
4         389  Afghanistan          AFG                 World   

  Counterpart-Area Code                                        Series Name  \
0                   WLD  Disbursements on external debt, long-term (DIS...   
1                   WLD  Disbursements on external debt, public and pub...   
2                   WLD                  PPG, bilateral (DIS, current US$)   
3                   WLD               PPG, multilateral (DIS, current US$)   
4                   WLD         PPG, official creditors (DIS, current US$)   

      Series Code  1970  1971  1972  ...        2019      2020        2021  \
0  DT.DIS.DLXF.CD   0.0   0.0   0.0  ...  28590201.1  440209.2  91525581.2   
1  DT.DIS.DPPG.CD   0.0   0.0   0.0  ...  28590201.1  440209.2  91525581.2   
2  DT.DIS.BLAT.CD   0.0   0.0   0.0  ...  27851000.5       0.0  81807974.7   
3  DT.DIS.MLAT.CD   0.0   0.0   0.0  ...    739200.6  440209.2   9717606.5   
4  DT.DIS.OFFT.CD   0.0   0.0   0.0  ...  28590201.1  440209.2  91525581.2   

          2022        2023        2024        2025        2026       2027  \
0  109228850.6  76137774.7  49745910.6  29280699.4  15282698.8  9068511.5   
1  109228850.6  76137774.7  49745910.6  29280699.4  15282698.8  9068511.5   
2   86924699.4  59607890.3  37990151.3  21443526.5   9922365.3  6005463.7   
3   22304151.2  16529884.4  11755759.3   7837172.9   5360333.5  3063047.8   
4  109228850.6  76137774.7  49745910.6  29280699.4  15282698.8  9068511.5   

        2028  
0  3491523.9  
1  3491523.9  
2  1960000.0  
3  1531523.9  
4  3491523.9  

[5 rows x 66 columns]

In [130]:
data['Country Name'].unique()

array(['Afghanistan', 'Albania', 'Algeria', 'Angola', 'Argentina',
       'Armenia', 'Azerbaijan', 'Bangladesh', 'Belarus', 'Belize',
       'Benin', 'Bhutan', 'Bolivia', 'Bosnia and Herzegovina', 'Botswana',
       'Brazil', 'Bulgaria', 'Burkina Faso', 'Burundi', 'Cabo Verde',
       'Cambodia', 'Cameroon', 'Central African Republic', 'Chad',
       'China', 'Colombia', 'Comoros', 'Congo, Dem. Rep.', 'Congo, Rep.',
       'Costa Rica', "Cote d'Ivoire", 'Djibouti', 'Dominica',
       'Dominican Republic',
       'East Asia & Pacific (excluding high income)', 'Ecuador',
       'Egypt, Arab Rep.', 'El Salvador', 'Eritrea', 'Eswatini',
       'Ethiopia', 'Europe & Central Asia (excluding high income)',
       'Fiji', 'Gabon', 'Gambia, The', 'Georgia', 'Ghana', 'Grenada',
       'Guatemala', 'Guinea', 'Guinea-Bissau', 'Guyana', 'Haiti',
       'Honduras', 'IDA only', 'India', 'Indonesia', 'Jamaica', 'Jordan',
       'Kazakhstan', 'Kenya', 'Kosovo', 'Kyrgyz Republic', 'Lao PDR',
       'Lat

In [133]:
data

Unnamed: 0 Country Name Country Code Counterpart-Area Name  \
0            110  Afghanistan          AFG                 World   
1            115  Afghanistan          AFG                 World   
2            338  Afghanistan          AFG                 World   
3            377  Afghanistan          AFG                 World   
4            389  Afghanistan          AFG                 World   
...          ...          ...          ...                   ...   
5630       77313     Zimbabwe          ZWE                 World   
5631       77341     Zimbabwe          ZWE                 World   
5632       77349     Zimbabwe          ZWE                 World   
5633       77352     Zimbabwe          ZWE                 World   
5634       77355     Zimbabwe          ZWE                 World   

     Counterpart-Area Code                                        Series Name  \
0                      WLD  Disbursements on external debt, long-term (DIS...   
1                      WLD  Disbursements on external debt, public and pub...   
2                      WLD                  PPG, bilateral (DIS, current US$)   
3                      WLD               PPG, multilateral (DIS, current US$)   
4                      WLD         PPG, official creditors (DIS, current US$)   
...                    ...                                                ...   
5630                   WLD                   PS, bilateral (TDS, current US$)   
5631                   WLD                PS, multilateral (TDS, current US$)   
5632                   WLD          PS, official creditors (AMT, current US$)   
5633                   WLD          PS, official creditors (INT, current US$)   
5634                   WLD          PS, official creditors (TDS, current US$)   

            Series Code       1970       1971       1972  ...        2019  \
0        DT.DIS.DLXF.CD        0.0        0.0        0.0  ...  28590201.1   
1        DT.DIS.DPPG.CD        0.0        0.0        0.0  ...  28590201.1   
2        DT.DIS.BLAT.CD        0.0        0.0        0.0  ...  27851000.5   
3        DT.DIS.MLAT.CD        0.0        0.0        0.0  ...    739200.6   
4        DT.DIS.OFFT.CD        0.0        0.0        0.0  ...  28590201.1   
...                 ...        ...        ...        ...  ...         ...   
5630  DT.TDS.BLAT.PS.CD  2925700.0  2867362.8  2409612.3  ...  47876686.1   
5631  DT.TDS.MLAT.PS.CD  5874000.0  5467000.0  7170000.0  ...   9532994.4   
5632  DT.AMT.OFFT.PS.CD  4587900.0  4242919.5  5140945.3  ...  45881749.7   
5633  DT.INT.OFFT.PS.CD  4211800.0  4091443.3  4438667.0  ...  11527930.8   
5634  DT.TDS.OFFT.PS.CD  8799700.0  8334362.8  9579612.3  ...  57409680.5   

            2020         2021         2022         2023         2024  \
0       440209.2   91525581.2  109228850.6   76137774.7   49745910.6   
1       440209.2   91525581.2  109228850.6   76137774.7   49745910.6   
2            0.0   81807974.7   86924699.4   59607890.3   37990151.3   
3       440209.2    9717606.5   22304151.2   16529884.4   11755759.3   
4       440209.2   91525581.2  109228850.6   76137774.7   49745910.6   
...          ...          ...          ...          ...          ...   
5630  29579518.8  138098223.5  141227325.8  129677924.0  127302273.9   
5631   4669355.8   29317983.1   32235172.3   33060159.4   32860559.8   
5632  17840378.4  132894125.1  130946083.6  118502605.9  116529303.8   
5633  16408496.2   34522081.5   42516414.5   44235477.5   43633529.9   
5634  34248874.6  167416206.6  173462498.1  162738083.4  160162833.7   

             2025         2026         2027         2028  
0      29280699.4   15282698.8    9068511.5    3491523.9  
1      29280699.4   15282698.8    9068511.5    3491523.9  
2      21443526.5    9922365.3    6005463.7    1960000.0  
3       7837172.9    5360333.5    3063047.8    1531523.9  
4      29280699.4   15282698.8    9068511.5    3491523.9  
...           ...          ...          ...          ...  
5630  1679120

In [134]:
data['Country Name']=data['Country Name'].replace(to_replace=['East Asia & Pacific (excluding high income)','Sub-Saharan Africa (excluding high income)','Europe & Central Asia (excluding high income)','Latin America & Caribbean (excluding high income)','Least developed countries: UN classification','Low & middle income','Low income','Lower middle income','Middle East & North Africa (excluding high income)','Middle income','Upper middle income'],value=np.nan)

In [138]:
data.dropna(inplace=True)

In [139]:
data_1=data.select_dtypes('float64')

In [140]:
data['Sum']=data_1.sum(axis=1)

In [141]:
b=pd.DataFrame(data.groupby('Country Name')['Sum'].sum())

In [142]:
b

Sum
Country Name               
Afghanistan    4.909411e+09
Albania        3.575531e+10
Algeria        1.995147e+12
Angola         4.022688e+11
Argentina      8.194342e+12
...                     ...
Venezuela, RB  5.185640e+12
Vietnam        4.613298e+11
Yemen, Rep.    5.100057e+10
Zambia         5.279659e+11
Zimbabwe       2.238202e+11

[121 rows x 1 columns]

In [146]:
b=b.sort_values(by='Sum',ascending=False)

In [147]:
b

Sum
Country Name                 
Brazil           2.737384e+13
Mexico           2.656635e+13
South Asia       2.541775e+13
India            1.698157e+13
Indonesia        1.631778e+13
...                       ...
Somalia          2.025441e+09
Vanuatu          1.982464e+09
Timor-Leste      1.935940e+09
Solomon Islands  1.935556e+09
Tonga            8.912793e+08

[121 rows x 1 columns]

In [148]:
b=b[0:9]

In [150]:
b.iplot(kind='bar',y='Sum',title='TOP 10 COUNTRIES WITH HIGEST DEPT WITHOUT GROUPING',colors='pink')

# SO ONE FIND THE TOP 10 COUNTRIES OVERVIEW (WITHOUT GROUPING)

In [5]:
data.head()

Unnamed: 0 Country Name Country Code Counterpart-Area Name  \
0         110  Afghanistan          AFG                 World   
1         115  Afghanistan          AFG                 World   
2         338  Afghanistan          AFG                 World   
3         377  Afghanistan          AFG                 World   
4         389  Afghanistan          AFG                 World   

  Counterpart-Area Code                                        Series Name  \
0                   WLD  Disbursements on external debt, long-term (DIS...   
1                   WLD  Disbursements on external debt, public and pub...   
2                   WLD                  PPG, bilateral (DIS, current US$)   
3                   WLD               PPG, multilateral (DIS, current US$)   
4                   WLD         PPG, official creditors (DIS, current US$)   

      Series Code  1970  1971  1972  ...        2019      2020        2021  \
0  DT.DIS.DLXF.CD   0.0   0.0   0.0  ...  28590201.1  440209.2  91525581.2   
1  DT.DIS.DPPG.CD   0.0   0.0   0.0  ...  28590201.1  440209.2  91525581.2   
2  DT.DIS.BLAT.CD   0.0   0.0   0.0  ...  27851000.5       0.0  81807974.7   
3  DT.DIS.MLAT.CD   0.0   0.0   0.0  ...    739200.6  440209.2   9717606.5   
4  DT.DIS.OFFT.CD   0.0   0.0   0.0  ...  28590201.1  440209.2  91525581.2   

          2022        2023        2024        2025        2026       2027  \
0  109228850.6  76137774.7  49745910.6  29280699.4  15282698.8  9068511.5   
1  109228850.6  76137774.7  49745910.6  29280699.4  15282698.8  9068511.5   
2   86924699.4  59607890.3  37990151.3  21443526.5   9922365.3  6005463.7   
3   22304151.2  16529884.4  11755759.3   7837172.9   5360333.5  3063047.8   
4  109228850.6  76137774.7  49745910.6  29280699.4  15282698.8  9068511.5   

        2028  
0  3491523.9  
1  3491523.9  
2  1960000.0  
3  1531523.9  
4  3491523.9  

[5 rows x 66 columns]